rust/signed_doc/src/validator/rules/content_type.rs

```diff
@@ -48,13 +48,93 @@ impl ContentTypeRule {
 
         Ok(true)
     }
+
+    /// Validates the provided `content` bytes to be a defined `ContentType`.
+    pub(crate) fn validate(&self, content: &[u8]) -> anyhow::Result<()> {
+        match self.exp {
+            ContentType::Json => {
+                if let Err(e) = serde_json::from_slice::<serde_json::Value>(content) {
+                    anyhow::bail!("Invalid {} content: {e}", self.exp)
+                }
+            },
+            ContentType::Cbor => {
+                let mut decoder = minicbor::Decoder::new(content);
+
+                decoder.skip()?;
+
+                if decoder.position() != content.len() {
+                    anyhow::bail!("Unused bytes remain in the input after decoding")
+                }
+            },
+        }
+        Ok(())
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use proptest::prelude::*;
+
     use super::*;
     use crate::Builder;
 
+    #[test]
+    fn content_type_validate_test() {
+        let json_rule = ContentTypeRule {
+            exp: ContentType::Json,
+        };
+        let cbor_rule = ContentTypeRule {
+            exp: ContentType::Cbor,
+        };
+
+        let json_bytes = serde_json::to_vec(&serde_json::Value::Null).unwrap();
+        assert!(json_rule.validate(&json_bytes).is_ok());
+        assert!(cbor_rule.validate(&json_bytes).is_err());
+
+        let cbor_bytes = minicbor::to_vec(minicbor::data::Token::Null).unwrap();
```

Comment:
We should not have multiple tests in the one test function where they are not logically related in a sequence.
This is two distinct tests, and should be two distinct test functions, because failure on either will fail the whole test, and CI won't disambiguate them, so we won't know one test passed and the next failed, or vice versa.